pkg/provisioner/ironic/ironic.go

```diff
@@ -825,6 +825,53 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node, data pr
 	return updater
 }
 
+// GetFirmwareSettings gets the BIOS settings and optional schema from the host and returns maps
+func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings metal3v1alpha1.SettingsMap, schema map[string]metal3v1alpha1.SettingSchema, err error) {
+
+	if p.nodeID == "" {
+		return nil, nil, provisioner.ErrNeedsRegistration
+	}
+
+	// Get the settings from Ironic via Gophercloud
+	var settingsList []nodes.BIOSSetting
+	var biosListErr error
+	if includeSchema {
```

Comment:
It seems like we more or less always need a schema, why having this switch?

Reply:
My thought here is once we have the schema it will no longer change so we just need to do a non-detailed query to get the settings without including the schema fields, reducing the API bandwidth.

Reply:
I wonder if the schema can change in run time, e.g. after a firmware update.

Reply:
Reading the updated patch it seems like includeSchema is always `true` now?